cmd/selfupdate.go

```diff
@@ -0,0 +1,257 @@
+	apiURL := fmt.Sprintf("https://api.github.com/repos/%s/%s/releases/latest", updateArgs.repoOwner, updateArgs.repoName)
+
+	resp, err := http.Get(apiURL)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	var releaseData struct {
+		TagName string `json:"tag_name"`
+	}
+	if err := json.NewDecoder(resp.Body).Decode(&releaseData); err != nil {
+		return "", err
+	}
+
+	return releaseData.TagName, nil
+}
+
+func determineBinaryName() string {
+	switch runtime.GOARCH {
+	case "amd64":
+		return "wings_linux_amd64"
+	case "arm64":
+		return "wings_linux_arm64"
+	default:
+		return ""
+	}
+}
+
+func downloadFile(url, fileName string) (string, error) {
+	tmpFile, err := os.CreateTemp("", fileName)
+	if err != nil {
+		return "", err
+	}
+	defer tmpFile.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("unexpected status: %s", resp.Status)
+	}
+
+	fmt.Printf("Downloading %s (%.2f MB)...\n", fileName, float64(resp.ContentLength)/1024/1024)
+	progressWriter := &progressWriter{Writer: tmpFile, Total: resp.ContentLength}
+	if _, err := io.Copy(progressWriter, resp.Body); err != nil {
+		return "", err
+	}
+
+	fmt.Println() // Ensure a newline after download progress
+	return tmpFile.Name(), nil
+}
+
+func verifyChecksum(binaryPath, checksumPath, binaryName string) error {
+	checksumData, err := os.ReadFile(checksumPath)
+	if err != nil {
+		return err
+	}
+
+	var expectedChecksum string
+	for _, line := range strings.Split(string(checksumData), "\n") {
+		if strings.HasSuffix(line, binaryName) {
+			parts := strings.Fields(line)
+			if len(parts) > 0 {
+				expectedChecksum = parts[0]
+			}
+			break
+		}
+	}
+	if expectedChecksum == "" {
+		return fmt.Errorf("checksum not found for %s", binaryName)
+	}
+
+	file, err := os.Open(binaryPath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	hasher := sha256.New()
+	if _, err := io.Copy(hasher, file); err != nil {
+		return err
+	}
+	actualChecksum := fmt.Sprintf("%x", hasher.Sum(nil))
+
+	if actualChecksum != expectedChecksum {
+		return fmt.Errorf("checksum mismatch: expected %s, got %s", expectedChecksum, actualChecksum)
+	}
+
+	return nil
+}
+
+func replaceBinary(currentPath, newPath string) error {
+	return os.Rename(newPath, currentPath)
+}
+
+type progressWriter struct {
+	io.Writer
+	Total     int64
+	Written   int64
+	StartTime time.Time
+}
+
+func (pw *progressWriter) Write(p []byte) (int, error) {
+	n, err := pw.Writer.Write(p)
+	pw.Written += int64(n)
+
+	if pw.Total > 0 {
+		percent := float64(pw.Written) / float64(pw.Total) * 100
+		fmt.Printf("\rProgress: %.2f%%", percent)
+	}
+
+	return n, err
+}
+
+func restartService() error {
+	// Try to run the systemctl restart command
+	cmd := exec.Command("systemctl", "restart", "wings")
```

Comment:
Breaks if the service is not named `wings` so it should be a config key

I see more and more machines having both ptero & pelican installed at the same time and that would restart the wrong wings too

Reply:
That 2 wings is installed wont be fixt as it is not supported to eun both ln the same machine. That none systemd one yes. We will just add it to the docs to tell the user to self restart it as there is no good way to do this in go.

Reply:
I will just trown that logic out.

Reply:
I just replaced this with a message that they have to restart the server themself